next-cloudinary/src/components/UploadWidget/UploadWidget.js

```diff
@@ -0,0 +1,103 @@
+import { useRef } from "react";
+import Script from "next/script";
+
+const UploadWidget = ({ children, onUpload, options, signed }) => {
+  const cloudinary = useRef();
+  const widget = useRef();
+
+  /**
+   * generateSignature
+   * @description Makes a request to an endpoint to sign Cloudinary parameters as part of widget creation
+   */
+
+  function generateSignature(callback, paramsToSign) {
```

Comment:
i'm thinking we're going to need to abstract this and actually avoid using a hard-coded endpoint, since it could vary depending on how someone wants to handle this

maybe we create a prop called `signingEndpoint` or something along those lines, and pass in the URL that way? that way they're not forced to create that specific URL. and document what is posted and what's expected to be returned